utils/i18nUtils.js

```diff
@@ -0,0 +1,103 @@
+import i18next from "i18next";
+import { initReactI18next } from "react-i18next";
+import ShopifyFormat from "@shopify/i18next-shopify";
+import resourcesToBackend from "i18next-resources-to-backend";
+import { match } from "@formatjs/intl-localematcher";
+import { shouldPolyfill } from "@formatjs/intl-locale/should-polyfill";
+import {
+  DEFAULT_LOCALE as DEFAULT_POLARIS_LOCALE,
+  SUPPORTED_LOCALES as SUPPORTED_POLARIS_LOCALES,
+} from "@shopify/polaris";
+
+// Update this to the default locale for the app
+const DEFAULT_APP_LOCALE = "en";
+// Update this to the supported locales contained in the `locales` folder
+const SUPPORTED_APP_LOCALES = ["en", "de", "fr"];
```

Comment:
Note: I considered scanning the `./locales` folder to determine the default locale and supported locales by file name so that it wouldn't be necessary to declare and maintain these constants, but I don't think this is something that should be done by the web runtime. However, I might consider doing something like this for the Remix app template, where the server can pass its computed values to the front end.